lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -885,13 +886,26 @@ class DecomposeAtenArgMaxOp : public OpRewritePattern<AtenArgmaxOp> {
       input = rewriter.create<AtenFlattenUsingIntsOp>(loc, flattenType, input,
                                                       dim, end);
     }
-    Value maxResult =
+
+    static_assert(std::is_same<OpTy, AtenArgmaxOp>() ||
+                  std::is_same<OpTy, AtenArgminOp>());
+    Value resultArg;
+    if constexpr (std::is_same<OpTy, AtenArgmaxOp>()) {
```

Comment:
I think a more typical way to do this is to template the conversion target as well, that way you don't need this constexpr. For example, see the implementation of the pooling lowerings: https://github.com/llvm/torch-mlir/blob/6244f301fb063be15dc5df27f77567fb8852e31f/lib/Conversion/TorchToLinalg/Pooling.cpp#L377

Reply:
Thanks, I already had the feeling there was a more elegant solution for this. Will change this